pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -113,7 +115,7 @@ public Consumer(Subscription subscription, SubType subType, long consumerId, int
         stats.clientVersion = cnx.getClientVersion();
 
         if (subType == SubType.Shared) {
-            this.pendingAcks = new ConcurrentOpenHashMap<PositionImpl, Integer>(256, 2);
+            this.pendingAcks = new ConcurrentLongLongPairHashMap(256, 2);
```

Comment:
I think we should change the map to use 1 single segment since it will not have a lot of contention.

`new ConcurrentLongLongPairHashMap(256, 1)`